src/main/scala/org/openhorizon/exchangeapi/route/apikey/UserApiKeys.scala

```diff
@@ -0,0 +1,274 @@
+       ).toResourceChangeRow else DBIO.successful(0)
+    } yield deleted).transactionally.asTry).map {
+    case Success(0) =>
+        (StatusCodes.NotFound, ApiResponse(ApiRespType.NOT_FOUND, ExchMsg.translate("apikey.not.found")))
+    case Success(_) =>
+         (StatusCodes.NoContent, ApiResponse(ApiRespType.OK, ExchMsg.translate("apikey.deleted")))
+    case Failure(ex) =>
+      logger.error(s"Error deleting API key $keyid", ex)
+       (StatusCodes.InternalServerError, ApiResponse(ApiRespType.INTERNAL_ERROR, ExchMsg.translate("apikey.deletion.failed")))
+  }
+  }
+
+  // === GET /v1/orgs/{organization}/users/{username}/apikeys/{keyid} ===
+  @GET
+  @Path("/{keyid}")
+  @Operation(
+  summary = "Get an API key by ID",
+  description = "Returns API key with the given ID. Must be called by the user on their own behalf (if they are the owner) or by an organization admin.",
+  parameters = Array(
+    new Parameter(name = "organization", in = ParameterIn.PATH, required = true, description = "Organization ID"),
+    new Parameter(name = "username", in = ParameterIn.PATH, required = true, description = "Username"),
+    new Parameter(name = "keyid", in = ParameterIn.PATH, required = true, description = "API key ID")
+  ),
+  responses = Array(
+    new responses.ApiResponse(
+      responseCode = "200",
+      description = "response body",
+      content = Array(new Content(
+        mediaType = "application/json",
+        schema = new Schema(implementation = classOf[ApiKeyMetadata]),
+        examples = Array(
+          new ExampleObject(
+            value = """{
+              "id": "string",
+              "description": "string",
+              "owner": "string",
+              "lastUpdated": "string"
+            }"""
+          )
+        )
+      ))
+    ),
+    new responses.ApiResponse(responseCode = "401", description = "invalid credentials"),
+    new responses.ApiResponse(responseCode = "403", description = "access denied"),
+    new responses.ApiResponse(responseCode = "404", description = "not found")
+  )
+)
+  def getUserApiKeyById(@Parameter(hidden = true)  identity: Identity2,
+                        @Parameter(hidden = true)  organization: String,
+                        @Parameter(hidden = true)  username: String,
+                        @Parameter(hidden = true)  keyid: UUID): Route = complete {
+
+    val userAndKeyQuery = for {
+      user <- UsersTQ.filter(u => u.organization === organization && u.username === username)
+      key  <- ApiKeysTQ.filter(k => k.id === keyid && k.user === user.user && k.orgid === organization)
+    } yield (user, key)
+
+    val fullQuery = for {
+      resultOpt   <- userAndKeyQuery.result.headOption
+      ownerStrOpt <- resultOpt match {
+                        case Some(_) => ApiKeysTQ.getOrgAndUsernameByKeyId(keyid)
+                        case None    => DBIO.successful(None)
+                      }
```

Comment:
You should already have owner id from the `resultOpt` - no need to fetch the the API key once more in `getOrgAndUsernameByKeyId`.